src/verifier/step4/EqPolynomial.sol

```diff
@@ -0,0 +1,161 @@
+pragma solidity ^0.8.0;
```

Comment:
Not worth spending time on this in the current PR, but:

My IDE gives a warning about no `SPDX license identifier`. It may be worth settling on one license and propagating through the repo (there are some inconsistencies about this currently).

Reply:
Right. Licensing information should be added, though I don't know what licence we will use. Probably, @huitseeker can advice here, and I can submit separate PR that will propagate it all over the codebase